src/Symfony/Component/Routing/Generator/UrlGeneratorInterface.php

```diff
@@ -83,4 +83,13 @@
      *                                             it does not match the requirement
      */
     public function generate($name, $parameters = array(), $referenceType = self::ABSOLUTE_PATH);
+
+    /**
+     * Returns whether a route with the given name exists.
+     *
+     * @param string $name The route name
+     *
+     * @return bool
+     */
+    public function hasRoute($name);
```

Comment:
@WouterJ, @fabpot I've removed the `hasRoute` method from the `UrlGeneratorInterface`